### ISSUE TITLE
Many non-damaging abilities cause a miniscule amount of threat.

### ISSUE THREAD

**Issue by Undertower** [chromiecraft/chromiecraft#1337]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [X] 40-49
- [ ] 50-59

### Current Behaviour

The list of abilities below causes threat:

Blood Rage
Rapid Fire
Aspect of the Hawk
Aspect of the Cheetah
Aspect of the Beast
Aspect of the Pack
Aspect of the Monkey
Track Humanoids
Track Undead
Track Hidden
Track Giants
Track Elementals
Track Demons
Track Beasts
Intimidation (Casting the ability causes threat to the hunter)
Eyes of the Beast
Eagle Eye
Scare Beast EDIT: Scare Beast is working as intended

### Expected Blizzlike Behaviour

None of the abilities listed should cause threat.

### Source

World of Warcraft: Classic and World of Warcraft: The Burning Crusade

### Steps to reproduce the problem

Have a pet, or a player, body pull a mob without causing any threat on the mob.
Use any of the abilities listed above.
Alternatively:
Download Omen3 Threat Meter
Aggro a mob
Use the various abilities to see your threat changing on the Threat Meter.

### Extra Notes

This is most notable on a hunter as you are questing and leveling, as you will constantly overaggro mobs your pet has aggroed without dealing damage or healing.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/3d4befd7043fbbceb32f7c2a6e0a4f9f638f8b0a

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-CustomWorldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by Azcobu** [comment]
Thanks for the report. This is an interesting one. I've tested with a hunter, and it worked exactly as you say above. I also tested with a warlock, and their utility buff abilities like Unending Breath, Sense Demons, and Shadow Ward also generated threat and pulled aggro from their pet. 

Did some checking and found this in a threat guide - "Buffs that are cast while in combat will generate threat that is distributed equally to all of the mobs that are actively engaged by the party/raid. Generally casting buffs generates very little threat." ([source](https://classic.wowhead.com/guides/threat-overview-classic-wow)) Which makes it sound as if this is working as expected.

So did some more checking with a few non-pet classes (druid, pally, warrior), and found that buffing a party member in combat generates threat, as expected, and will pull aggro. But buffing yourself doesn't. As a pally I could stand there and buff myself all day and the mobs wouldn't aggro onto me. So there's this odd difference, where a pet class in combat can buff themselves and pull threat, but another class can buff themselves when a party member is in combat and have no effect. 

I don't think it's because hunter and warlock buffs are treated differently to every other class's buffs. I think it may have to do with how a pet in combat is treated vs how a party member in combat is treated. If a pet class's pet gets into combat, that puts the hunter/lock into combat as well, so their self buffs generate threat. But parties work differently, where if a party member is in combat and has aggro, another party member is not automatically pulled into combat as well, and so as a non-combatant, their self buffs don't generate threat.

So to summarise, I think this may be expected behaviour (odd as it seems) and you just don't see it in any non-pet classes because they'll always have aggro in a fight anyway. But more research is needed, and I'll leave this open for further comments or suggestions.

**Comment by Undertower** [comment]
There never was an issue with threat in this way in retail Wrath of the Lich King, and there isn't any issue with this in Classic or Classic TBC. The buffs I mentioned are all self buffs(Intimidation included), maybe other buffs such as Paladin Blessings, Priest Prayers, Druid Mark will cause threat. But never once have the buffs I mentioned above caused threat in any expansion I have played. I am currently logged onto classic and classic tbc and will record some proof for you.

**Comment by Undertower** [comment]
Here's a video from Classic:
https://www.youtube.com/watch?v=k8jthEOl0MA
Here's a video from Classic TBC:
https://www.youtube.com/watch?v=rG0tMZlAkIE

As you can see, none of the abilities I tested cause any threat to the hunter.
The only ability that did cause threat was the Scare Beast, and I have updated the original post accordingly.

**Comment by Azcobu** [comment]
Thanks for the videos, that looks like decent evidence that AC isn't behaving correctly.
